src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
+        if (size == elementData.length) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
 
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("The index passed to method is invalid");
+        }
+        if (size == elementData.length) {
+            grow();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
 
     }
 
+    private void grow() {
+        Object[] newList = new Object[(int) (size * 1.5)];
+        System.arraycopy(elementData, 0, newList, 0, size);
+        elementData = newList;
+    }
+
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
```

Comment:
You can create a void method to check if the index is between 0 and size, and use it in all other checks.